### PR TITLE
Make finalize-script optional for file recordings

### PR DESCRIPTION
**Pull request by thisisnkc**
## Make finalize-script optional

The `finalize-script` in `reference.conf` had a dummy default (`/path/to/finalize`) which meant every deployment had to override it, even if they didn't need a finalize script at all.

This changes it so the property is optional i.e, if you don't set it, Jibri just skips finalization and moves on. Empty strings are also handled the same way.

No breaking changes, existing setups that already configure a finalize script will work exactly as before.

### CODE REVIEW

File: src/main/kotlin/org/jitsi/jibri/service/impl/FileRecordingJibriService.kt [hunk 1]
```diff
@@ -116,10 +117,7 @@ class FileRecordingJibriService(
         "JibriConfig::recordingDirectory" { Config.legacyConfigSource.recordingDirectory!! }
         "jibri.recording.recordings-directory".from(Config.configSource)
     }
-    private val finalizeScriptPath: String by config {
-        "JibriConfig::finalizeRecordingScriptPath" {
-            Config.legacyConfigSource.finalizeRecordingScriptPath!!
-        }
+    private val finalizeScriptPath: String? by optionalconfig {
```
Comment (saghul) [inline src/main/kotlin/org/jitsi/jibri/service/impl/FileRecordingJibriService.kt:120]: You are not handling the backwards compat here, are you?

**Comment by saghul** [comment]
The idea SGTM, can't comment much on the code.

File: src/main/kotlin/org/jitsi/jibri/service/impl/FileRecordingJibriService.kt (hunk 1 above)
Comment (thisisnkc) [reply src/main/kotlin/org/jitsi/jibri/service/impl/FileRecordingJibriService.kt:120]: Good catch, thanks! I missed the legacy config source when switching to optionalconfig. Fixed now.

**Comment by thisisnkc** [comment]
I missed on the backward compatibility, should we add tests for that as I can see there are none currently?
